src/components/deploymentConfig/DeploymentTemplateView.tsx

```diff
@@ -0,0 +1,831 @@
+            setGroupedOptions([
+                {
+                    label: '',
+                    options: [baseTemplateOption],
+                },
+                {
+                    label: 'values on other environments',
+                    options: environments.length > 0 ? environments : [{ label: 'No options', value: 0, info: '' }],
+                },
+            ])
+
+            if (!selectedOption) {
+                setSelectedOption(baseTemplateOption)
+            }
+        } else {
+            setGroupedOptions([
+                {
+                    label: '',
+                    options: environments,
+                },
+            ])
+
+            if (!selectedOption) {
+                setSelectedOption(environments[0])
+            }
+        }
+    }, [environments])
+
+    const onChange = (selected: DeploymentChartOptionType) => {
+        setSelectedOption(selected)
+    }
+
+    return (
+        <ReactSelect
+            options={groupedOptions}
+            isMulti={false}
+            value={selectedOption}
+            classNamePrefix="compare-template-values-select"
+            formatOptionLabel={formatOptionLabel}
+            isOptionDisabled={(option) => option.value === 0}
+            isSearchable={false}
+            onChange={onChange}
+            components={{
+                IndicatorSeparator: null,
+                Option,
+                DropdownIndicator,
+                ValueContainer: customValueContainer,
+            }}
+            styles={{
+                control: (base) => ({
+                    ...base,
+                    backgroundColor: 'var(--N100)',
+                    border: 'none',
+                    boxShadow: 'none',
+                    minHeight: '32px',
+                }),
+                option: (base, state) => ({
+                    ...base,
+                    color: 'var(--N900)',
+                    backgroundColor: state.isFocused ? 'var(--N100)' : 'white',
+                }),
+                menu: (base) => ({
+                    ...base,
+                    marginTop: '2px',
+                    minWidth: '240px',
+                }),
+                menuList: (base) => ({
+                    ...base,
+                    position: 'relative',
+                    paddingBottom: 0,
+                    paddingTop: 0,
+                    maxHeight: '250px',
+                }),
+                dropdownIndicator: (base, state) => ({
+                    ...base,
+                    padding: 0,
+                    color: 'var(--N400)',
+                    transition: 'all .2s ease',
+                    transform: state.selectProps.menuIsOpen ? 'rotate(180deg)' : 'rotate(0deg)',
+                }),
+                noOptionsMessage: (base) => ({
+                    ...base,
+                    color: 'var(--N600)',
+                }),
+            }}
+        />
+    )
+}
+
+const getCodeEditorHeight = (
+    isUnSet: boolean,
```

Comment:
same here

Reply:
Done

Reply:
It's a normal function :)

Reply:
But have provided the type for the component below this function.